lib/util/lock.ml

```diff
@@ -25,7 +25,7 @@ let control ~onerror lname wait f =
          let r = f () in Unix.close fd ; r
        with e -> Unix.close fd ; raise e)
     with
-    | Unix.Unix_error _ -> onerror ()
+    | (Unix.Unix_error _) as e -> print_endline ("Unix Error :" ^ Printexc.to_string e); onerror ()
     | e -> raise e
```

Comment:
I need to clean this, that print was only for testing purposes. We can use result instead of raising exceptions, the rest of the code is not working this way, so we would end up raising exceptions at call site I guess.